src/app/composer/qgscomposer.cpp

```diff
@@ -2136,6 +2136,7 @@ void QgsComposer::exportCompositionAsImage( QgsComposer::OutputMode mode )
     QFileDialog dlg( this, tr( "Export atlas to directory" ) );
     dlg.setFileMode( QFileDialog::Directory );
     dlg.setOption( QFileDialog::ShowDirsOnly, true );
+    dlg.setOption( QFileDialog::DontUseNativeDialog, true );
```

Comment:
@luipir , check this paragraph on the QFileDialog documentation (http://doc.qt.io/qt-5/qfiledialog.html):

_By default, a platform-native file dialog will be used if the platform has one. In that case, the widgets which would otherwise be used to construct the dialog will not be instantiated, so related accessors such as layout() and itemDelegate() will return null. You can set the DontUseNativeDialog option to ensure that the widget-based implementation will be used instead of the native dialog._

In this case, our code calls layout(), which crashes QGIS since layout()'s pointer is null.